db/migrate/20170905210626_create_configuration_templates.rb

```diff
@@ -0,0 +1,12 @@
+class CreateConfigurationTemplates < ActiveRecord::Migration[5.0]
+  def change
+    create_table :configuration_templates do |t|
+      t.bigint :ems_id
+      t.string :external_id
```

Comment:
The _external_id_ field is used to represent the unique ID that's generated by LXCA for each configuration pattern.

Reply:
fields ending in `_id` are reserved for ActiveRecord relations.   Since this is a unique id, let's use ems_ref so that it matches all of the other tables, since that is the "native unique id"

Reply:
@Fryguy So are you suggesting to change _external_id_ to _ems_ref_? The _external_id_ field is meant to represent the unique ID that is generated by LXCA for each config pattern. We are storing this in the DB, so when the user selects a particular pattern, we can retrieve and send the correct ID to the LXCA REST interface to execute the pattern. Just making sure I understand because I notice that _ems_ref_ is used in other tables and thought it was for MIQ-specific purposes. Thanks

Reply:
Correct.  ems_ref is the *native* unique identifier, so that sounds exactly like what you are describing.